### PR TITLE
[WIP] implement backup item actio plugin for volmesnapshotcontents

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Fixes https://github.com/vmware-tanzu/velero/issues/2371

Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

### CODE REVIEW

File: internal/backup/csi_volumesnapshot.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		vs.Labels = make(map[string]string)
+	}
+
+	for k, v := range vals {
+		vs.Annotations[k] = v
+		vs.Labels[k] = v
+	}
+}
+
+// Execute backs VolumeSnapshotContents associated with the volumesnapshot as additional resource.
```
Comment (nrb) [inline internal/backup/csi_volumesnapshot.go]: Is this file added because it's based on PR #31?

Comment (ashish-amarnath) [reply internal/backup/csi_volumesnapshot.go]: Yes, I cut a branch for this work from the branch in PR #31 
Sorry if that's what caused the confusion

**Comment by skriss** [comment]
Changes this PR currently contains:

- add BackupItemAction for VolumeSnapshotContent that:
    - adds deletion secret, if it exists, as additional item to back up
- modify RestoreItemAction for VolumeSnapshotContent to:
    - not modify the VolumeSnapshotContent spec
    - add deletion secret, if it exists, as additional item to restore

**Comment by ashish-amarnath** [comment]
This PR is made obsolete by PR #34 which covers secrets for volumesnapshotcontents and volumesnapshotclass. Closing this in favor of #34

**Pull request closed without merging**